SignalServiceKit/src/Messages/OWSMessageReceiver.m

```diff
@@ -220,16 +222,34 @@ - (void)enqueueEnvelopeForProcessing:(OWSSignalServiceProtosEnvelope *)envelope
 
 - (void)drainQueue
 {
-    dispatch_async(self.class.serialGCDQueue, ^{
-        OWSMessageProcessingJob *_Nullable job = [self.finder nextJob];
-        if (job == nil) {
-            DDLogVerbose(@"%@ Queue is drained", self.tag);
-            return;
-        }
+    AssertIsOnMainThread();
+
+    if (self.isDrainingQueue) {
```

Comment:
I'm not clear on why this was necessary. `serialGCDQueue` already ensured that only one job was being parsed at a time - is that we were queuing up a ton of work on `serialGCDQueue`?  Shouldn't that be cheap?

Put another way: can we now get rid of `serialGCDQueue` if we're going to use a flag to ensure that only one job is processing at a time?